api/v1beta2/bgpadvertisement_types.go

```diff
@@ -0,0 +1,80 @@
+/*
+
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta2
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required. Any new fields you add must have json tags for the fields to be serialized.
+
+// BGPAdvertisementSpec defines the desired state of BGPAdvertisement.
+type BGPAdvertisementSpec struct {
+	// The aggregation-length advertisement option lets you “roll up” the /32s into a larger prefix.
+	// +kubebuilder:validation:Minimum=1
+	// +kubebuilder:default:=32
+	// +optional
+	AggregationLength *int32 `json:"aggregationLength,omitempty"`
+
+	// Optional, defaults to 128 (i.e. no aggregation) if not
+	// specified.
+	// +kubebuilder:default:=128
+	// +optional
+	AggregationLengthV6 *int32 `json:"aggregationLengthV6,omitempty"`
+
+	// BGP LOCAL_PREF attribute which is used by BGP best path algorithm,
+	// Path with higher localpref is preferred over one with lower localpref.
+	LocalPref uint32 `json:"localPref,omitempty"`
+
+	// BGP communities
+	Communities []string `json:"communities,omitempty"`
+
+	// AddressPools is the list of addresspools to advertise via this advertisement.
+	AddressPools []string `json:"addressPools,omitempty"`
+}
+
+// BGPAdvertisementStatus defines the observed state of BGPAdvertisement.
+type BGPAdvertisementStatus struct {
+	// INSERT ADDITIONAL STATUS FIELD - define observed state of cluster
```

Comment:
is that a TODO?

Reply:
It's generated by controller-gen, and it was the same in the operator.